Signal/src/ViewControllers/Utils/MessageRecipientStatusUtils.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc enum MessageRecipientStatus: Int {
+    case uploading
+    case sending
+    case sent
+    case delivered
+    case read
+    case failed
+}
+
+class MessageRecipientStatusUtils: NSObject {
```

Comment:
We show message recipient status in more than one view; this class gathers those concerns.